src/main/java/org/hiero/sketch/table/ArrayRowOrder.java

```diff
@@ -0,0 +1,41 @@
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.table.api.IRowIterator;
+import org.hiero.sketch.table.api.IRowOrder;
+
+import java.util.ArrayList;
+
+public class ArrayRowOrder implements IRowOrder {
```

Comment:
This class needs a JavaDoc comment. The name of the class could be misinterpreted in many ways.
Perhaps you should call it ExplicitRowOrder.